price-feeder/oracle/oracle.go

```diff
@@ -228,18 +243,24 @@ func (o *Oracle) SetPrices(ctx context.Context, acceptList oracletypes.DenomList
 		}
 	}
 
-	filteredProviderPrices, err := o.filterDeviations(providerPrices)
+	tvwapPrices, err := ComputeTVWAP(providerCandles)
```

Comment:
I think this whole section needs a pretty elaborate godoc comment explaining what we're doing here. It's getting a bit complex IMO.